chromium_src/components/permissions/permission_request.h

```diff
@@ -34,6 +34,9 @@ class PermissionRequest : public PermissionRequest_ChromiumImpl {
   void SetLifetime(absl::optional<base::TimeDelta> lifetime);
   const absl::optional<base::TimeDelta>& GetLifetime() const;
 
+  void SetDontAskAgain(bool dont_ask_again);
```

Comment:
these are specific to widevine, why are you adding it to the generic PermissionRequest?

Reply:
It's @goodov 's suggestion to solve the problem of layer violation, see https://github.com/brave/brave-core/pull/18958#discussion_r1274812925.

Reply:
@goodov couldn't we instead move `WidevinePermissionRequest` into components by passing `chrome::AttemptRelaunch` and `EnableWidevineCdmComponent` as callbacks in the constructor?

Reply:
`DontAskWidevineInstall` could be moved to a components version of widevine_utils

Reply:
I asked him about your suggestion before and he said it better to generalize in case we want another permission with this "don't ask" logic.